### PR TITLE
New: On Movie Delete and On Movie File Delete Notifications

### LINKED ISSUE #5812: Pull Sonarr commit 'New: On Delete Notifications' [Multiple]
Pull https://github.com/Sonarr/Sonarr/commit/c91fabcf2d14afda7dc689c8e2a73edba50c2639

```
git cherry-pick -ex c91fabcf2d14afda7dc689c8e2a73edba50c2639
```

This commit has 15 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
YES 191

#### Description
Pull of On Delete Notifications as implemented in Sonarr.

#### Todos
- [x] Tests
- [x] Translation Keys
- [x] Wiki Updates

#### Issues Fixed or Closed by this PR

* Fixes #5812

### CODE REVIEW

**Comment by RobinDadswell** [comment]
Included issues #5816, #5817 and #5818

**Comment by RobinDadswell** [comment]
@bakerboy448 is there an easy way to copy the Sonarr wiki entry

**Comment by RobinDadswell** [comment]
Wiki is also now updated - https://wiki.servarr.com/Settings_Connect#Radarr

**Comment by Qstick** [comment]
Looks good overall, I'll do a deep dive on it tomorrow or Friday likely

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 1]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
         </div>
 
         {
-          supportsOnGrab && onGrab &&
+          supportsOnGrab && onGrab ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnGrab')}
-            </Label>
+              {translate('OnGrab')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:89]: Get rid of the `;` it comes through literal in js to the label

image

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 2]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            null
         }
 
         {
-          supportsOnDelete && onDelete &&
+          supportsOnDownload && onDownload ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnDelete')}
-            </Label>
+              {translate('OnImport')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:97]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 3]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            null
         }
 
         {
-          supportsOnDownload && onDownload &&
+          supportsOnUpgrade && onDownload && onUpgrade ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnImport')}
-            </Label>
+              {translate('OnUpgrade')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:105]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 4]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            null
         }
 
         {
-          supportsOnUpgrade && onDownload && onUpgrade &&
+          supportsOnRename && onRename ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnUpgrade')}
-            </Label>
+              {translate('OnRename')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:113]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 5]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            null
         }
 
         {
-          supportsOnRename && onRename &&
+          supportsOnHealthIssue && onHealthIssue ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnRename')}
-            </Label>
+              {translate('OnHealthIssue')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:121]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 6]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            null
         }
 
         {
-          supportsOnHealthIssue && onHealthIssue &&
+          supportsOnMovieDelete && onMovieDelete ?
             <Label kind={kinds.SUCCESS}>
-              {translate('OnHealthIssue')}
-            </Label>
+              {translate('OnMovieDelete')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:129]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 7]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            </Label> :
+            null
         }
 
         {
-          !onGrab && !onDownload && !onRename && !onHealthIssue && !onDelete &&
+          supportsOnMovieFileDelete && onMovieFileDelete ?
+            <Label kind={kinds.SUCCESS}>
+              {translate('OnMovieFileDelete')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:137]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 8]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+            </Label> :
+            null
+        }
+
+        {
+          supportsOnMovieFileDeleteForUpgrade && onMovieFileDelete && onMovieFileDeleteForUpgrade ?
+            <Label kind={kinds.SUCCESS}>
+              {translate('OnMovieFileDeleteForUpgrade')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:145]: Same `;`

File: frontend/src/Settings/Notifications/Notifications/Notification.js [hunk 9]
```diff
@@ -80,55 +84,78 @@ class Notification extends Component {
...
+
+        {
+          !onGrab && !onDownload && !onRename && !onHealthIssue && !onMovieDelete && !onMovieFileDelete ?
             <Label
               kind={kinds.DISABLED}
               outline={true}
             >
-              {translate('Disabled')}
-            </Label>
+              {translate('Disabled')};
```
Comment (Qstick) [inline frontend/src/Settings/Notifications/Notifications/Notification.js:156]: Same `;`

**Review by Qstick — CHANGES_REQUESTED** [review]

File: frontend/src/Settings/Notifications/Notifications/Notification.js (hunk 1 above)
Comment (austinwbest) [reply frontend/src/Settings/Notifications/Notifications/Notification.js:89]: Looking at that image, OnUpgrade is not translated or something is wrong with the key it seems. Or the translation has no space

File: frontend/src/Settings/Notifications/Notifications/Notification.js (hunk 2 above)
Comment (RobinDadswell) [reply frontend/src/Settings/Notifications/Notifications/Notification.js:97]: Done

File: frontend/src/Settings/Notifications/Notifications/Notification.js (hunk 1 above)
Comment (RobinDadswell) [reply frontend/src/Settings/Notifications/Notifications/Notification.js]: Yeah the translation key was missing! Added it

**Comment by RobinDadswell** [comment]
Fixed and squashed...

**Review by Qstick — APPROVED** [review]

**Comment by RobinDadswell** [comment]
@Qstick apologies I have added the triggers into discord notifier so it got rid of your review

**Review by Qstick — APPROVED** [review]

**Pull request merged**